pkg/codegen/nodejs/gen.go

```diff
@@ -1157,7 +1157,7 @@ func (mod *modContext) genIndex(exports []string) string {
 		if mod.compatibility == kubernetes20 {
 			// Trim version suffix from child modules. Nested versions will have their own index.ts file.
 			if match := k8sVersionSuffix.FindStringIndex(child); len(match) != 0 {
-				child = child[:match[0]]
+				child = child[match[0]:match[1]]
```

Comment:
With the suggestion above, this would be:

```suggestion
				child = child[match[2]:match[3]]
```

`match[2:3]` contains the bounds for the first capturing group.